marklogic-client-api/src/test/java/com/marklogic/client/test/LineSplitterTest.java

```diff
@@ -0,0 +1,100 @@
+
+    @Before
+    public void setUp() {
+        client = DatabaseClientFactory.newClient("localhost", 8012,
+                new DatabaseClientFactory.DigestAuthContext("rest-admin", "x"));
+    }
+
+    @Test
+    public void testSplitter() throws Exception {
+        LineSplitter splitter = new LineSplitter();
+        Stream<StringHandle> contentStream = splitter.split(new FileInputStream(jsonlFile));
+        assertNotNull(contentStream);
+
+        StringHandle[] result = contentStream.toArray(size -> new StringHandle[size]);
+        assertNotNull(result);
+        assertEquals(result.length, Files.lines(Paths.get(jsonlFile)).count());
+
+        String[] originalResult = Files.lines(Paths.get(jsonlFile))
+                                    .toArray(size -> new String[size]);
+
+        for (int i = 0; i < result.length && i < originalResult.length; i++) {
+            assertNotNull(result[i].get());
+            assertEquals(result[i].get(), originalResult[i]);
+        }
+    }
+
+    @Test
+    public void testSplitterGzip() throws Exception {
+        LineSplitter splitter = new LineSplitter();
+        GZIPInputStream gzipStream = new GZIPInputStream(new FileInputStream(jsonlGzipFile));
+        Stream<StringHandle> contentStream = splitter.split(gzipStream);
+        assertNotNull(contentStream);
+
+        StringHandle[] result = contentStream.toArray(size -> new StringHandle[size]);
+        assertNotNull(result);
+
+        gzipStream = new GZIPInputStream(new FileInputStream(jsonlGzipFile));
+        String[] originalResult = new BufferedReader(new InputStreamReader(gzipStream))
+                                    .lines()
+                                    .toArray(size -> new String[size]);
+        assertEquals(result.length, originalResult.length);
+
+        for (int i = 0; i < result.length && i < originalResult.length; i++) {
+            assertNotNull(result[i].get());
+            assertEquals(result[i].get(), originalResult[i]);
+        }
+    }
+
+    @Test
+    public void testSplitterXML() throws Exception {
+        LineSplitter splitter = new LineSplitter();
+        splitter.setFormat(Format.XML);
+        Stream<StringHandle> contentStream = splitter.split(new FileInputStream(xmlFile));
+        assertNotNull(contentStream);
+
+        StringHandle[] result = contentStream.toArray(size -> new StringHandle[size]);
+        assertNotNull(result);
+        assertEquals(result.length, Files.lines(Paths.get(xmlFile)).count());
+
+        String[] originalResult = Files.lines(Paths.get(xmlFile))
+                                    .toArray(size -> new String[size]);
+        assertEquals(result[0].getFormat(), Format.XML);
+        for (int i = 0; i < result.length && i < originalResult.length; i++) {
+            assertNotNull(result[i].get());
+            assertEquals(result[i].get(), originalResult[i]);
+        }
+    }
+
+    @After
+    public void closeSetUp() {
+        client.release();
```

Comment:
If the test uses com.marklogic.client.test.Common.connect(), there's no need to release -- other tests can use the same connection.

Reply:
I removed the After part as there's no need for a connection.